src/Core/Commander/Providers/WMTS_Provider.js

```diff
@@ -109,147 +86,69 @@ WMTS_Provider.prototype.url = function url(coWMTS, layer) {
  * @param {type} coWMTS : coord WMTS
  * @returns {WMTS_Provider_L15.WMTS_Provider.prototype@pro;_IoDriver@call;read@call;then}
  */
-WMTS_Provider.prototype.getXbilTexture = function getXbilTexture(tile, layer, parameters) {
-    var cooWMTS = tile.wmtsCoords[layer.options.tileMatrixSet][0];
-    var pitch = new THREE.Vector3(0.0, 0.0, 1.0);
-
-    if (parameters.ancestor) {
-        // account for possible level offset between coords and tile level
-        // (e.g for PM texture cooWMTS.level = tile.level + 1)
-        var levelOffset = cooWMTS.zoom - tile.level;
-
-        cooWMTS = this.projection.WMTS_WGS84Parent(
-            cooWMTS,
-            this.computeLevelToDownload(tile, parameters.ancestor, layer) + levelOffset,
+WMTS_Provider.prototype.getXbilTexture = function getXbilTexture(tile, layer, parentTextures) {
+    const coordWMTS = tile.wmtsCoords[layer.options.tileMatrixSet][0];
+    const pitch = new THREE.Vector3(0.0, 0.0, 1.0);
+
+    if (parentTextures) {
+        WMTS_WGS84Parent(
```

Comment:
Well all right. I see. This WMTS_WGS84Parent function is really bad. It has side-effects AND returns a value. This is very confusing and error-prone. I'm not requiring anything here as it is just a copy-paste, but we might want to refacto that in the future.